### PR TITLE
feat: trim metadatas (part 2)

### PR DESCRIPTION
**Pull request by lwwmanning**
now all of the encodings have PTypes instead of DTypes, removed unnecessary lengths, etc.

also renamed e.g., `zigzag.rs` to `array.rs` to standardize the per-encoding file layout

### CODE REVIEW

File: encodings/fsst/src/array.rs [hunk 1]
```diff
@@ -80,17 +82,17 @@ impl FSSTArray {
 
         let symbols_len = symbols.len();
         let len = codes.len();
-        let strings_dtype = codes.dtype().clone();
-        let uncompressed_lengths_dtype = uncompressed_lengths.dtype().clone();
+        let uncompressed_lengths_ptype = uncompressed_lengths.dtype().try_into()?;
```
Comment (danking) [inline encodings/fsst/src/array.rs]: Should we prefer the more explicit `PType::try_from(...)`

File: encodings/runend/src/array.rs [hunk 2]
```diff
@@ -148,14 +148,16 @@ impl RunEndArray {
...
-                self.metadata().num_runs,
-            )
+            .child(0, &self.ends_dtype(), self.metadata().num_runs)
             .vortex_expect("RunEndArray is missing its run ends")
     }
 
+    /// Get the DType of the ends array
+    #[inline]
+    pub fn ends_dtype(&self) -> DType {
+        DType::Primitive(self.metadata().ends_ptype, Nullability::NonNullable)
```
Comment (danking) [inline encodings/runend/src/array.rs]: This is `DType::from(self.metadata().ends_ptype)`

**Review by danking — DISMISSED** [review]

**Comment by robert3005** [comment]
Somehow you managed to type coerce values in forarray

**Comment by lwwmanning** [comment]
> Somehow you managed to type coerce values in forarray

FlexBuffers got me, they automatically narrow the integer type when writing, and we don't store the PValue enum type because we assumed they'd round trip (or that we'd cast to the dtype)

https://github.com/spiraldb/vortex/blob/develop/vortex-scalar/src/serde/serde.rs

**Review by robert3005 — APPROVED** [review]

**Pull request merged**